pkg/cmd/api/api.go

```diff
@@ -0,0 +1,122 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/buildkite/cli/v3/pkg/cmd/factory"
+	"github.com/buildkite/cli/v3/pkg/cmd/validation"
+	"github.com/spf13/cobra"
+)
+
+var (
+	method      string
+	headers     []string
+	data        string
+	baseURL     string
+	graphqlFlag bool
+	queryFile   string
+)
+
+func NewCmdAPI(f *factory.Factory) *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "api <endpoint>",
+		Short: "Interact with the Buildkite API",
+		Long:  "Interact with either the REST or GraphQL Buildkite APIs",
+		Args:  cobra.MaximumNArgs(1),
+		Example: heredoc.Doc(`
+      # To make an API call
+      $ bk api pipelines/example-pipeline/builds/420
+      `),
+		PersistentPreRunE: validation.CheckValidConfiguration(f.Config),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return apiCaller(cmd, args, f)
+		},
+	}
+
+	cmd.Flags().StringVarP(&method, "method", "X", "GET", "HTTP method to use")
+	cmd.Flags().StringArrayVarP(&headers, "header", "H", []string{}, "Headers to include in the request")
+	cmd.Flags().StringVarP(&data, "data", "d", "", "Data to send in the request body")
+	cmd.Flags().StringVar(&baseURL, "base-url", fmt.Sprintf("https://api.buildkite.com/v2/organizations/%s", f.Config.OrganizationSlug()), "Base URL for the API")
+	cmd.Flags().BoolVar(&graphqlFlag, "graphql", false, "Use GraphQL API")
+	cmd.Flags().StringVarP(&queryFile, "file", "f", "", "File containing GraphQL query")
+
+	return &cmd
+}
+
+func apiCaller(cmd *cobra.Command, args []string, f *factory.Factory) error {
```

Comment:
I know that this is just an initial work, but just wondering if we are going to move this to the same resolver models that was started for the other cli commands?

Reply:
@lizrabuya IMO, which I'm happy to have changed, the intent of the `api` command is to return API responses too, the API would return a JSON blob and so that should be the value returned by this command.

Reply:
I don't see a use for the build resolver things in this part of the CLI

Reply:
I think allowing things to be filled in from the current repo would be nice, but not necessary at this point. Ie `/pipelines/:pipeline/builds` or something which would resolve the current directories repo/pipeline

Reply:
yeah, maybe we visit that later. It's too early to decide right now. Thanks both!